### ISSUE TITLE
Process fails to start on node command with arguments

### ISSUE THREAD

**Issue by dtitenko** [adamralph/simple-exec#119]
Attempting to run `yarn install` command `Run("yarn", "install")`
Results in
```
cmd.exe /c ""yarn" install"
module.js:550
    throw err;
    ^

Error: Cannot find module 'C:\Projects\Project1\src\node_modules\yarn\bin\yarn.js'
```
Actual command should be `cmd.exe /c "yarn install"`.

**Comment by dtitenko** [comment]
Similarly, `Run("yarn", null)` results in
```
cmd.exe /c ""yarn" "
module.js:550
    throw err;
    ^

Error: Cannot find module 'C:\Projects\PermitVision\src\node_modules\yarn\bin\yarn.js'
```
where should've been `cmd.exe /c "yarn"`.

**Comment by adamralph** [comment]
Hey @dtitenko thanks for raising this and the related PR. I'm on the road right now but I'll look into this ASAP.

**Comment by dtitenko** [comment]
Found a workaround.
`await RunAsync($"{nodeExePath}", $"{yarnCliPath} {command}");`
None will install yarn globally anyway. Using [Yarn.MSbuild](https://www.nuget.org/packages/Yarn.MSBuild/) process can start from tools with args.
Can be closed.

**Comment by damianh** [comment]
Hey, just discovered this with `5.0.1` which emits `cmd.exe /c ""npm" install --scripts-prepend-node-path"` resulting in a

```
internal/modules/cjs/loader.js:651
    throw err;
    ^

Error: Cannot find module 'E:\dev\{hidden}\src\js\test-app\node_modules\npm\bin\npm-cli.js'
```


Reverting back to 4.2.0 and using a previous workaround:

```csharp
        private static void RunCmd(string name, string args = null, string workingDirectory = null, bool noEcho = false)
        {
            try
            {
                Run(name, args, workingDirectory, noEcho);
            }
            catch (Win32Exception ex) when (ex.NativeErrorCode == 2) // The system cannot find the file specified.
            {
                var cmdArgs = $"/C {name}";
                if (args != null)
                {
                    cmdArgs += " " + args;
                }
                Run("cmd", cmdArgs, workingDirectory, noEcho);
            }
        }
```

...which emits `cmd.exe /c npm install --scripts-prepend-node-path` and which works.

Now while it works, I don't know if it has other side affects or not as it appears the `"` impacts things.

**Comment by damianh** [comment]
🤔 Wondering if running with `cmd` should be opt-in?

**Comment by adamralph** [comment]
@damianh do you know what the underlying problem is? I.e. Why `cmd.exe /c ""npm" install --scripts-prepend-node-path"` fails?

**Comment by damianh** [comment]
Sorry I don't. @dtitenko has this in his report:

`Error: Cannot find module 'C:\Projects\Project1\src\node_modules\yarn\bin\yarn.js'`

For some reason it is looking for yarn in this path idea

I tried `cmd.exe /c ""npm" install --scripts-prepend-node-path"` and `cmd.exe /c npm install --scripts-prepend-node-path` from the CLI and they _both_ worked ffs.

**Comment by pawelros** [comment]
I had exactly the same problem while trying to execute `npm install` from Bullseye. The workaround that worked for me:
`
 public npm_install(params string[] dependsOn)
        {
            Target(
                nameof(npm_install),
                DependsOn(dependsOn),
                () =>
                {
                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                    {
                        Run("cmd", "/c npm install", Path.Combine(Directory.GetCurrentDirectory(), "ui"));
                    }
                    else
                    {
                        Run("npm", "install", Path.Combine(Directory.GetCurrentDirectory(), "ui"));
                    }
                });
        }
`

https://github.com/nodejs/help/issues/1271 I believe relates to the same problem but none of the solutions mentioned there worked for me.

**Comment by adamralph** [comment]
Thanks for the info @pawelros. I'm glad you have a workaround for now.

FYI this issue is currently top of my list to look at, as soon as I get time.

**Comment by adamralph** [comment]
FTR the currently proposed solution for this is #128

**Comment by adamralph** [comment]
Marten is now using #128 to run it's [npm commands](https://github.com/JasperFx/marten/blob/1fc125856fdf92db1227e4aa1ef7506bdc66f4a7/martenbuild.cs#L28-L32) using the [new arguments](https://github.com/JasperFx/marten/blob/1fc125856fdf92db1227e4aa1ef7506bdc66f4a7/martenbuild.cs#L129), so it seems like a workable solution.

#128 is available in [6.0.0-beta.1]. To follow the progress towards 6.0.0 RTM, please follow the [release issue](https://github.com/adamralph/simple-exec/issues/129).
